examples/11_hens/utilities.py

```diff
@@ -0,0 +1,1050 @@
+
+    lon_range_model : int
+        The range of the input longitude coordinates (either 180 or 360).
+
+    lon_range_out : int
+        The range of the desired output longitude coordinates (either 180 or 360).
+
+    Returns
+    -------
+    lat_coords_sub : np.ndarray[np.float64]
+        A 1D array containing the latitude values of the subgrid, ordered descending.
+
+    lon_coords_sub : np.ndarray[np.float64]
+        A 1D array containing the longitude values of the subgrid, ordered ascending.
+
+    Raises
+    ------
+    ValueError
+        If the longitude range conversion is not supported.
+
+    Notes
+    -----
+    - The function handles the conversion between 180-degree and 360-degree longitude ranges.
+    - The function filters the latitude and longitude coordinates to fit within the specified cropbox.
+    """
+    if lon_range_model != lon_range_out:
+        # determine where to split for conversion
+        if lon_range_model == 180 and lon_range_out == 360:
+            split_longitude = 0
+        elif lon_range_model == 360 and lon_range_out == 180:
+            split_longitude = 180
+
+        ind_below = (lon_coords < split_longitude).nonzero()[0]
+        ind_above = (lon_coords >= split_longitude).nonzero()[0]
+        if lon_range_model == 180 and lon_range_out == 360:
+            lon_coords = np.concatenate(
+                [lon_coords[ind_above], lon_coords[ind_below] + 360]
+            )
+        elif lon_range_model == 360 and lon_range_out == 180:
+            lon_coords = np.concatenate(
+                [lon_coords[ind_above] - 360, lon_coords[ind_below]]
+            )
+
+    # filter/crop area
+    idx_lon = np.where(
+        (lon_coords >= cfg_cropbox.lon_min) & (lon_coords <= cfg_cropbox.lon_max)
+    )
+    idx_lat = np.where(
+        (lat_coords >= cfg_cropbox.lat_min) & (lat_coords <= cfg_cropbox.lat_max)
+    )
+    lat_coords_sub = lat_coords[min(idx_lat[0]) : max(idx_lat[0]) + 1]
+    lon_coords_sub = lon_coords[min(idx_lon[0]) : max(idx_lon[0]) + 1]
+    return lat_coords_sub, lon_coords_sub
+
+
+def update_model_dict(model_dict: dict, package: Package) -> dict:
+    """
+    check if model on GPU is same as needed for next inference.
+    If not, load new model package and update model dict.
+
+    Parameters
+    ----------
+    model_dict : dict
+        dictionary specifying model, model class, and model package
+    package : Package
+        model package to be used in next inference
+
+    Returns
+    -------
+    dict
+        model dict.
+    """
+    if package != model_dict["package"]:
+        model_dict["package"] = package
+        package = (
+            model_dict["class"].load_default_package()
+            if package == "default"
+            else Package(package)
+        )
+        model_dict["model"] = model_dict["class"].load_model(package=package)
+
+    return model_dict
+
+
+def store_tracks(area_name: str, tracks: list[pd.DataFrame], cfg: DictConfig) -> None:
```

Comment:
My suggestions with the tracks is just convert the tensor and coords from the tracker into a xarray data array with the NaNs, then just save it all the netCDF. No need to convert into some data frame. Let the plotting functions deal with the filtering / mapping each point to a timestep.

Seems like a tough data structure to save... since theres potential holes in the paths.

Reply:
Updated this myself